website/handlers/router.ts

```diff
@@ -166,29 +176,37 @@ export default function RoutesSelection(
 
     const timing = monitoring?.timings.start("router");
 
-    const routesFromProps = Array.isArray(audiences) ? audiences : [];
-    // everyone should come first in the list given that we override the everyone value with the upcoming flags.
-    const [routes, hrefRoutes] = buildRoutes(
-      Array.isArray(ctx.routes)
-        ? [...ctx.routes, ...routesFromProps]
-        : routesFromProps,
-    );
-    // build the router from entries
-    const builtRoutes = Object.entries(routes).sort((
-      [routeStringA, { highPriority: highPriorityA }],
-      [routeStringB, { highPriority: highPriorityB }],
-    ) =>
-      (highPriorityB ? HIGH_PRIORITY_ROUTE_RANK_BASE_VALUE : 0) +
-      rankRoute(routeStringB) -
-      ((highPriorityA ? HIGH_PRIORITY_ROUTE_RANK_BASE_VALUE : 0) +
-        rankRoute(routeStringA))
-    );
-
+    const prepareRoutes = () => {
+      const routesFromProps = Array.isArray(audiences) ? audiences : [];
+      // everyone should come first in the list given that we override the everyone value with the upcoming flags.
+      const [routes, hrefRoutes] = buildRoutes(
+        Array.isArray(ctx.routes)
+          ? [...ctx.routes, ...routesFromProps]
+          : routesFromProps,
+      );
+      // build the router from entries
+      const builtRoutes = Object.entries(routes).sort((
+        [routeStringA, { highPriority: highPriorityA }],
+        [routeStringB, { highPriority: highPriorityB }],
+      ) =>
+        (highPriorityB ? HIGH_PRIORITY_ROUTE_RANK_BASE_VALUE : 0) +
+        rankRoute(routeStringB) -
+        ((highPriorityA ? HIGH_PRIORITY_ROUTE_RANK_BASE_VALUE : 0) +
+          rankRoute(routeStringA))
+      );
+      return {
+        routes: builtRoutes.map((route) => ({
+          pathTemplate: route[0],
+          handler: { value: route[1].func },
+        })),
+        hrefRoutes,
+      };
+    };
+    const routerId = `${RouterId.fromFlags(ctx.flags)}/${ctx.revision ?? ""}`;
+    routerCache[routerId] ??= prepareRoutes();
+    const { routes, hrefRoutes } = routerCache[routerId];
```

Comment:
Maybe routerCache should be an weakLFU, since the number of flagIds may have an exponential relation with the number of flags.

Just want to avoid an unbalanced tradeoff between performance and memory.